scripts/new_upstream_snapshot.py

```diff
@@ -333,14 +334,30 @@ def get_bugs_fixed_devel():
             yield line.split("LP: #")[1].strip()
 
 
+def increment_major_version(original: str) -> str:
+    """Given a version number, increment and return the major version.
+
+    Examples:
+    22.1 -> 22.2
+    22.2.3 -> 22.3
+    22.3.4~literally-anything -> 22.4
+    22.4 -> 23.1
+    """
+    split = original.split(".", 1)
+    major = int(split[0])
+    minor = int(split[1][0]) + 1
+    if minor == 5:
```

Comment:
I guess we don't really have to care here as we are unlikely to release greater than quarterly anyway. But, couldn't we just check if two digit major match todays year. If not, increment major and set minor to 1.